compiler/src/iree/compiler/Codegen/SPIRV/SPIRVGeneralizeNamedOps.cpp

```diff
@@ -0,0 +1,58 @@
+// Copyright 2023 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+//===- SPIRVGeneralizeNamedOps.cpp - Pass to generalize named linalg ops --===//
+//
+// The pass is to generalize named linalg ops that are better as linalg.generic
+// ops in IREE
+//
+//===----------------------------------------------------------------------===//
+
+#include "iree/compiler/Codegen/SPIRV/PassDetail.h"
+#include "iree/compiler/Codegen/SPIRV/Passes.h"
+#include "mlir/Dialect/Linalg/IR/Linalg.h"
+#include "mlir/Dialect/Linalg/Transforms/Transforms.h"
+#include "mlir/Pass/Pass.h"
+
+namespace mlir {
+namespace iree_compiler {
+
+namespace {
+struct SPIRVGeneralizeNamedOpsPass
+    : public SPIRVGeneralizeNamedOpsBase<SPIRVGeneralizeNamedOpsPass> {
+
+  void runOnOperation() override;
+};
+} // namespace
+
+void SPIRVGeneralizeNamedOpsPass::runOnOperation() {
+  auto funcOp = getOperation();
+  SmallVector<linalg::MatmulTransposeBOp> namedOpCandidates;
+  funcOp.walk([&](linalg::MatmulTransposeBOp linalgOp) {
+    if (isa_and_nonnull<linalg::MatmulTransposeBOp>(linalgOp.getOperation())) {
+      namedOpCandidates.push_back(linalgOp);
+    }
```

Comment:
I think we don't need `isa_and_nonnull`?  The walk method only walk through `MatmulTransposeBOp` op if type is specified.